src/gc/proxy.d

```diff
@@ -1,118 +1,53 @@
 
-/*          Copyright Digital Mars 2005 - 2013.
+/*          Copyright Digital Mars 2005 - 2016.
  * Distributed under the Boost Software License, Version 1.0.
  *    (See accompanying file LICENSE or copy at
  *          http://www.boost.org/LICENSE_1_0.txt)
  */
 module gc.proxy;
 
-import gc.gc;
+import gc.impl.conservative.gc;
+import gc.impl.manual.gc;
+import gc.config;
 import gc.stats;
-import core.stdc.stdlib;
+import gc.gcinterface;
+
 
 private
 {
-    __gshared GC _gc;
-
     static import core.memory;
     alias BlkInfo = core.memory.GC.BlkInfo;
 
     extern (C) void thread_init();
     extern (C) void thread_term();
 
-    struct Proxy
-    {
-        extern (C)
-        {
-            void function() gc_enable;
-            void function() gc_disable;
-        nothrow:
-            void function() gc_collect;
-            void function() gc_minimize;
-
-            uint function(void*) gc_getAttr;
-            uint function(void*, uint) gc_setAttr;
-            uint function(void*, uint) gc_clrAttr;
-
-            void*   function(size_t, uint, const TypeInfo) gc_malloc;
-            BlkInfo function(size_t, uint, const TypeInfo) gc_qalloc;
-            void*   function(size_t, uint, const TypeInfo) gc_calloc;
-            void*   function(void*, size_t, uint ba, const TypeInfo) gc_realloc;
-            size_t  function(void*, size_t, size_t, const TypeInfo) gc_extend;
-            size_t  function(size_t) gc_reserve;
-            void    function(void*) gc_free;
-
-            void*   function(void*) gc_addrOf;
-            size_t  function(void*) gc_sizeOf;
-
-            BlkInfo function(void*) gc_query;
-
-            void function(void*) gc_addRoot;
-            void function(void*, size_t, const TypeInfo ti) gc_addRange;
-
-            void function(void*) gc_removeRoot;
-            void function(void*) gc_removeRange;
-            void function(in void[]) gc_runFinalizers;
-
-            bool function() gc_inFinalizer;
-        }
-    }
-
-    __gshared Proxy  pthis;
-    __gshared Proxy* proxy;
-
-    void initProxy()
-    {
-        pthis.gc_enable = &gc_enable;
-        pthis.gc_disable = &gc_disable;
-        pthis.gc_collect = &gc_collect;
-        pthis.gc_minimize = &gc_minimize;
-
-        pthis.gc_getAttr = &gc_getAttr;
-        pthis.gc_setAttr = &gc_setAttr;
-        pthis.gc_clrAttr = &gc_clrAttr;
-
-        pthis.gc_malloc = &gc_malloc;
-        pthis.gc_qalloc = &gc_qalloc;
-        pthis.gc_calloc = &gc_calloc;
-        pthis.gc_realloc = &gc_realloc;
-        pthis.gc_extend = &gc_extend;
-        pthis.gc_reserve = &gc_reserve;
-        pthis.gc_free = &gc_free;
-
-        pthis.gc_addrOf = &gc_addrOf;
-        pthis.gc_sizeOf = &gc_sizeOf;
-
-        pthis.gc_query = &gc_query;
-
-        pthis.gc_addRoot = &gc_addRoot;
-        pthis.gc_addRange = &gc_addRange;
+    __gshared GC currentGC;  //used for making the GC calls
+    __gshared GC initialGC; //used to reset currentGC if gc_clrProxy was called
```

Comment:
You're right it's needed for the whacky Windows DLL support, I renamed and added comments.
https://github.com/Jebbs/druntime/pull/2/commits/8f99b220b4b22c1041df2d6107f450d8a0a4a68d